## MARKETSTACK

In [23]:
import requests

# API endpoint
base_url = 'http://api.marketstack.com/v2/eod'

# Parameters for the request
params = {
    'access_key': API_KEY,
    'symbols': 'AAPL',
    'date_from': '2025-02-13',
    'date_to': '2025-02-23'
}

# Make the request
response = requests.get(base_url, params=params)

# Get the JSON data
if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print(f"Error: {response.status_code}")

{'pagination': {'limit': 100, 'offset': 0, 'count': 6, 'total': 6}, 'data': [{'open': 245.95, 'high': 248.69, 'low': 245.22, 'close': 245.55, 'volume': 53119400.0, 'adj_high': 248.69, 'adj_low': 245.22, 'adj_close': 245.55, 'adj_open': 245.95, 'adj_volume': 53197431.0, 'split_factor': 1.0, 'dividend': 0.0, 'name': 'Apple Inc', 'exchange_code': 'NASDAQ', 'asset_type': 'Stock', 'price_currency': 'usd', 'symbol': 'AAPL', 'exchange': 'XNAS', 'date': '2025-02-21T00:00:00+0000'}, {'open': 244.97, 'high': 246.78, 'low': 244.29, 'close': 245.83, 'volume': 31125149.0, 'adj_high': 246.78, 'adj_low': 244.29, 'adj_close': 245.83, 'adj_open': 244.94, 'adj_volume': 32316907.0, 'split_factor': 1.0, 'dividend': 0.0, 'name': 'Apple Inc', 'exchange_code': 'NASDAQ', 'asset_type': 'Stock', 'price_currency': 'usd', 'symbol': 'AAPL', 'exchange': 'XNAS', 'date': '2025-02-20T00:00:00+0000'}, {'open': 244.66, 'high': 246.01, 'low': 243.16, 'close': 244.87, 'volume': 32184200.0, 'adj_high': 246.01, 'adj_low': 2

In [24]:
# API endpoint
base_url = 'http://api.marketstack.com/v2/tickerinfo'

# Parameters for the request
params = {
    'access_key': API_KEY,
    'ticker': 'TSLA'
}

# Make the request
response = requests.get(base_url, params=params)

# Get the JSON data
if response.status_code == 200:
    data = response.json()
    print(data)
else:
    print(f"Error: {response.status_code}")

{'data': {'name': 'Tesla, Inc.', 'ticker': 'TSLA', 'item_type': 'equity', 'sector': 'Consumer Cyclical', 'industry': 'Auto Manufacturers', 'exchange_code': 'NMS', 'full_time_employees': '127855', 'ipo_date': None, 'date_founded': None, 'key_executives': [{'name': 'Mr. Rodney D. Westmoreland Jr.', 'salary': '', 'function': 'Director of Construction Management', 'exercised': '', 'birth_year': ''}, {'name': 'Mr. Turner  Caldwell', 'salary': '', 'function': 'Engineering Manager', 'exercised': '', 'birth_year': ''}, {'name': 'Travis  Axelrod', 'salary': '', 'function': 'Head of Investor Relations', 'exercised': '', 'birth_year': ''}, {'name': 'Mr. Xiaotong  Zhu', 'salary': '926.88k', 'function': 'Senior Vice President of Automotive', 'exercised': '', 'birth_year': '1980'}, {'name': 'Mr. Vaibhav  Taneja', 'salary': '278k', 'function': 'Chief Financial Officer', 'exercised': '', 'birth_year': '1978'}, {'name': 'Mr. Elon R. Musk', 'salary': '', 'function': 'Co-Founder, Technoking of Tesla, CEO

In [ ]:
# Get first 10 tickers from the DataFrame
first_10_tickers = df['Ticker'].head(10).tolist()

# Create an empty list to store results
api_results = []

for ticker in first_10_tickers:
    # Parameters for the request
    params = {
        'access_key': API_KEY,
        'symbols': ticker
    }
    
    # Make the request
    response = requests.get('http://api.marketstack.com/v2/tickerslist', params=params)
    
    if response.status_code == 200:
        api_results.append(response.json())
    else:
        print(f"Error for {ticker}: {response.status_code}")

# Print results
for result in api_results:
    print(result)

## FINNHUB

In [29]:
!pip install finnhub-python
import finnhub


[notice] A new release of pip is available: 25.0 -> 25.0.1
[notice] To update, run: pip install --upgrade pip


In [ ]:
finnhub_client = finnhub.Client(api_key="")

print(finnhub_client.quote('TSLA'))


{'c': 337.8, 'd': -16.6, 'dp': -4.684, 'h': 354.98, 'l': 334.42, 'o': 353.44, 'pc': 354.4, 't': 1740258000}


- c: 337.8: Current price ($337.80).
- d: -16.6: Dollar change from previous close (-$16.60).
- dp: -4.684: Percentage change from previous close (-4.684%).
- h: 354.98: Day’s high price ($354.98).
- l: 334.42: Day’s low price ($334.42).
- o: 353.44: Opening price ($353.44).
- pc: 354.4: Previous close price ($354.40).
- t: 1740258000: Timestamp (October 23, 2024, 17:00 UTC).

Change the display on streamlit to make it more readable

## ALPHAVANTAGE

In [2]:
import requests

# replace the "demo" apikey below with your own key from https://www.alphavantage.co/support/#api-key
url = 'https://www.alphavantage.co/query?function=GLOBAL_QUOTE&symbol=TSLA&apikey=api_key'
r = requests.get(url)
data = r.json()

print(data)

{'Global Quote': {'01. symbol': 'TSLA', '02. open': '353.4400', '03. high': '354.9800', '04. low': '334.4200', '05. price': '337.8000', '06. volume': '74058648', '07. latest trading day': '2025-02-21', '08. previous close': '354.4000', '09. change': '-16.6000', '10. change percent': '-4.6840%'}}


## COINCAP

In [8]:
import requests

url = "https://api.coincap.io/v2/assets/bitcoin"

payload={}
headers = {}

response = requests.request("GET", url, headers=headers, data=payload)

print(response.text)


{"data":{"id":"bitcoin","rank":"1","symbol":"BTC","name":"Bitcoin","supply":"19828478.0000000000000000","maxSupply":"21000000.0000000000000000","marketCapUsd":"1877509486064.5287860103889952","volumeUsd24Hr":"7772814984.1104750523238259","priceUsd":"94687.5239776108275184","changePercent24Hr":"-0.8566179739135940","vwap24Hr":"95425.1496016107966084","explorer":"https://blockchain.info/"},"timestamp":1740415168140}


## COINPAPRIKA

In [6]:
!pip install coinpaprika-sdk


[notice] A new release of pip is available: 25.0 -> 25.0.1
[notice] To update, run: pip install --upgrade pip


In [7]:
from coinpaprika import client as Coinpaprika

client = Coinpaprika.Client()